plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -347,6 +383,33 @@ export class GitLabClient {
       } as PagedResponse<any>;
     });
   }
+
+  async nonPagedRequest<T = any>(
+    endpoint: string,
+    options?: CommonListOptions,
+  ): Promise<T> {
+    const request = new URL(`${this.config.apiBaseUrl}${endpoint}`);
+    for (const key in options) {
+      if (options[key] !== undefined && options[key] !== '') {
+        request.searchParams.append(key, options[key]!.toString());
```

Comment:
```suggestion
        request.searchParams.append(key, options[key].toString());
```

Reply:
I would leave this one as it is because we're already checking if the options exist in line 393 ( and line 358, where the same construction was used before). If I remove the non-null assertion operator, even if the check has been done before, Typescript warns me that the "Object is possibly 'undefined'". 
Is there any other way around this?

Reply:
ah good point I think something like this should work:

```js
    for (const key in options) {
      const value = options[key];
      if (value !== undefined && value !== '') {
        request.searchParams.append(key, value.toString());
      }
    }
```

so that we are more protected in case the `if` gets removed in the future

Reply:
Done!